esp-ieee802154/src/compat/mod.rs

```diff
@@ -0,0 +1,130 @@
+use core::{ffi::VaListImpl, fmt::Write};
+
+use log::info;
+
+use self::str_buf::StrBuf;
+
+mod str_buf;
+
+#[no_mangle]
+pub unsafe extern "C" fn phy_printf(format: *const u8, args: ...) {
+    syslog(format, args);
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn rtc_printf(format: *const u8, args: ...) {
+    syslog(format, args);
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn coexist_printf(format: *const u8, args: ...) {
+    syslog(format, args);
+}
+
+pub unsafe extern "C" fn syslog(format: *const u8, args: VaListImpl) {
+    let mut buf = [0u8; 512];
+    vsnprintf(&mut buf as *mut u8, 511, format, args);
+    let res_str = StrBuf::from(&buf as *const u8);
+    info!("{}", res_str.as_str_ref());
+}
+
+pub(crate) unsafe fn vsnprintf(
```

Comment:
@bjoernQ Don't we have this in esp-wifi too? Maybe we can share impls at some point 🤔

Reply:
Yes, we probably can/want to have something like a libc-compat crate - probably best to do once both crates are moved to the esp-hal repo